boleto/src/main/java/com/adyen/checkout/boleto/internal/ui/DefaultBoletoDelegate.kt

```diff
@@ -0,0 +1,301 @@
+        countryOptions: List<AddressListItem> = emptyList(),
+        stateOptions: List<AddressListItem> = emptyList(),
+    ): BoletoOutputData {
+        val updatedCountryOptions = AddressFormUtils.markAddressListItemSelected(
+            countryOptions,
+            inputData.address.country
+        )
+        val updatedStateOptions = AddressFormUtils.markAddressListItemSelected(
+            stateOptions,
+            inputData.address.stateOrProvince
+        )
+
+        val addressFormUIState = AddressFormUIState.fromAddressParams(componentParams.addressParams)
+
+        return BoletoOutputData(
+            firstNameState = BoletoValidationUtils.validateFirstName(inputData.firstName),
+            lastNameState = BoletoValidationUtils.validateLastName(inputData.lastName),
+            socialSecurityNumberState = SocialSecurityNumberUtils.validateSocialSecurityNumber(
+                inputData.socialSecurityNumber
+            ),
+            addressState = AddressValidationUtils.validateAddressInput(
+                inputData.address,
+                addressFormUIState,
+                updatedCountryOptions,
+                updatedStateOptions,
+                false
+            ),
+            addressUIState = addressFormUIState,
+            isSendEmailSelected = inputData.isSendEmailSelected,
+            shopperEmailState = BoletoValidationUtils.validateShopperEmail(
+                inputData.isSendEmailSelected,
+                inputData.shopperEmail
+            )
+        )
+    }
+
+    @VisibleForTesting
+    internal fun updateComponentState(outputData: BoletoOutputData) {
+        Logger.v(TAG, "updateComponentState")
+        val componentState = createComponentState(outputData)
+        _componentStateFlow.tryEmit(componentState)
+    }
+
+    private fun createComponentState(
+        outputData: BoletoOutputData = this.outputData
+    ): BoletoComponentState {
+        val paymentComponentData = PaymentComponentData(
+            paymentMethod = GenericPaymentMethod(paymentMethod.type),
+            order = order,
+            socialSecurityNumber = outputData.socialSecurityNumberState.value,
+            shopperName = ShopperName(
+                firstName = outputData.firstNameState.value,
+                lastName = outputData.lastNameState.value
+            )
+        )
+        if (outputData.isSendEmailSelected) {
+            paymentComponentData.shopperEmail = outputData.shopperEmailState.value
+        }
+        if (AddressFormUtils.isAddressRequired(outputData.addressUIState)) {
+            paymentComponentData.billingAddress = AddressFormUtils.makeAddressData(
+                addressOutputData = outputData.addressState,
+                addressFormUIState = outputData.addressUIState
+            )
+        }
+        return BoletoComponentState(
+            data = paymentComponentData,
+            isInputValid = outputData.isValid,
+            isReady = true
```

Comment:
Same thing as before: are we sure the component is finished with loading the address data?